packages/shared/src/react/contexts.tsx

```diff
@@ -1,28 +1,54 @@
 'use client';
 
 import type { PropsWithChildren } from 'react';
-import React from 'react';
+import React, { useState } from 'react';
 
 import type {
   BillingSubscriptionPlanPeriod,
   ClerkOptions,
-  ClientResource,
   ForPayerType,
+  InitialState,
   LoadedClerk,
   OrganizationResource,
-  SignedInSessionResource,
-  UserResource,
 } from '../types';
+import { useOrganizationBase } from './hooks/base/useOrganizationBase';
 import { createContextAndHook } from './hooks/createContextAndHook';
-import { SWRConfigCompat } from './providers/SWRConfigCompat';
 
 const [ClerkInstanceContext, useClerkInstanceContext] = createContextAndHook<LoadedClerk>('ClerkInstanceContext');
-const [UserContext, useUserContext] = createContextAndHook<UserResource | null | undefined>('UserContext');
-const [ClientContext, useClientContext] = createContextAndHook<ClientResource | null | undefined>('ClientContext');
-const [SessionContext, useSessionContext] = createContextAndHook<SignedInSessionResource | null | undefined>(
-  'SessionContext',
+export { useUserBase as useUserContext } from './hooks/base/useUserBase';
+export { useClientBase as useClientContext } from './hooks/base/useClientBase';
+export { useSessionBase as useSessionContext } from './hooks/base/useSessionBase';
+
+const [InitialStateContext, _useInitialStateContext] = createContextAndHook<InitialState | undefined>(
+  'InitialStateContext',
 );
 
+export function InitialStateProvider({
```

Comment:
I just noticed I had forgotten to write a JSDoc comment for this. This is a new public API community SDK packages could use.